### ISSUE TITLE
Allow setting a cookie for ecosystem preference

### ISSUE THREAD

**Issue by strugee** [thangngoc89/openit.io#7]
Tbh I don't actually need this myself since I'm an npm person anyway, but I feel bad that other ecosystems don't get the sweet "top-level" treatment that JS does where the app responds to `/*`. Seems like it'd be neat if you could configure settings or something on the homepage for your preferred ecosystem.

**Comment by thangngoc89** [comment]
Hi @strugee 

I've been thinking about this lately. The problem is I'm caching everything at CDN level up to 1 day.
Reason behind it:

- Reduce the load on my tiny server 
- Reduce API requests to repository's API
- Keep everything stateless

**Comment by strugee** [comment]
Ahhhh right. That makes sense.

You _could_ still keep everything stateless this way... don't set a session cookie, just literally set the preferred ecosystem value in the cookie value. Well, I guess it's not strictly speaking stateless, but you're pushing all the state to the browser ;)

I wonder if you could just serve a 3xx response on `/*` to `/:ecosystem/:package`? I've no idea what your usage looks like, obviously, so I don't know if that would still be too much load. But that _would_ allow you to continue caching the "expensive" queries.

**Comment by thangngoc89** [comment]
@strugee I guess I have to check with Cloudflare (since It's my current CDN). 

But the idea of redirect  `/*` to `/:ecosystem/:package` is interesting. It wouldn't really hit any expensive external API. Thank you :100:

**Comment by strugee** [comment]
@thangngoc89 definitely! I'm curious to know how this turns out - though as I said, I'm fine with the default ;)
